test/zksync-tests/EIP712.test.ts

```diff
@@ -0,0 +1,62 @@
+import {expect} from "chai";
+import {ethers} from "ethers";
+import {deployContract, provider} from "./shared/zkSyncUtils";
+import {MockPermit2WithCustomChainID, Permit2} from "../../typechain-types";
+import {Wallet} from "zksync-web3";
+
+
+describe("EIP712", function () {
+
+    const TYPE_HASH = ethers.utils.keccak256(ethers.utils.toUtf8Bytes("EIP712Domain(string name,uint256 chainId,address verifyingContract)"));
+    const NAME_HASH = ethers.utils.keccak256(ethers.utils.toUtf8Bytes("Permit2"));
+
+    let permit2: Permit2;
+    let permit2CustomChainID: MockPermit2WithCustomChainID;
+    let owner:Wallet = new Wallet('0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110', provider);
+    describe("Test Domain Separator", function () {
+
+        beforeEach('Deploy Permit2', async () => {
+            permit2 = await deployContract('Permit2') as Permit2;
+            permit2CustomChainID = await deployContract('MockPermit2WithCustomChainID') as MockPermit2WithCustomChainID;
```

Comment:
I am not sure you need to use `MockPermit2WithCustomChainID`, you can use `MockEIP712WithCustomChainID`